### PR TITLE
[14.0][MIG] report_label

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

**Review by ivantodorovich — APPROVED** [review]
Code review LGTM :+1: 
gh is complaining about pre-commit though, would you check?

**Comment by StefanRijnhart** [comment]
@ivantodorovich fixed now. Thanks for the nice module BTW!

**Comment by ivantodorovich** [comment]
> @ivantodorovich fixed now. Thanks for the nice module BTW!

Thank you for migrating it :heart:

File: report_label/__manifest__.py
Comment (rousseldenis) [inline report_label/__manifest__.py]: @StefanRijnhart I have a problem launching an update --all after having installed report_label.

Adding 'base_automation' here solves the problem, but I haven't found the root problem.

image

Comment (StefanRijnhart) [reply report_label/__manifest__.py]: Thanks, I'll have a look

Comment (StefanRijnhart) [reply report_label/__manifest__.py]: There you have it: https://github.com/odoo/odoo/blob/14.0/addons/base_automation/views/base_automation_view.xml#L8-L9. Base automation extends the view from `ir.actions.server` but `report_label`'s button is invalid for that model unless `report_label` depends on base_automation so that the button's action is also available on the `base.automation` model (which also inherits from the `ir.actions.server` model, by delegation)
I added the module dependency because I can't think of a less hacky way around it at this moment.

**Review by Daniel-CA — APPROVED** [review]
LGTM

**Review by andreampiovesana — APPROVED** [review]
go

**Comment by elvise** [comment]
@StefanRijnhart can you rebase please ?

**Comment by elvise** [comment]
@rousseldenis merge ?

**Comment by StefanRijnhart** [comment]
@elvise rebase done.

**Comment by francesco-ooops** [comment]
@StefanRijnhart error in precommit prevents module installation in runboat, would you have some time to try to fix it?
let us know if you need any help.

**Comment by StefanRijnhart** [comment]
Thanks for noting, @francesco-ooops! Fixed now.

**Comment by elvise** [comment]
@StefanRijnhart Great ! 😊

**Comment by elvise** [comment]
Hi @StefanRijnhart checked in the runboat but looks missing the menu 
IMG_1097

**Comment by StefanRijnhart** [comment]
It's marked for installation in the runboat logs, but the logs stop after installing module 39 out of 48: https://runboat.odoo-community.org/api/v1/builds/b99a74bb9-617a-4279-bd2f-2aae739565d9/init-log. Do you know what could have caused that?

**Comment by francesco-ooops** [comment]
> It's marked for installation in the runboat logs, but the logs stop after installing module 39 out of 48: https://runboat.odoo-community.org/api/v1/builds/b99a74bb9-617a-4279-bd2f-2aae739565d9/init-log. Do you know what could have caused that?

@sbidoul any hint on this behavior?

**Comment by francesco-ooops** [comment]
seems runboat can't load correctly module kpi_dashboard_bokeh in any PR in this repo after it was merged [here](https://github.com/OCA/reporting-engine/pull/506), @etobella do you have any idea what could be the cause?

**Comment by sbidoul** [comment]
I see no additional information on the cluster. So it looks like the installation indeed exited with code 0 before having finished... this might be reproducible locally.

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
on the functional side: I followed configuration + usage instructions and tried to use demo data from the module itself, but after launching the server action a document layout page that seems out of context pops up, and label is not printed

not sure if the problem lies with demo data or code as I cannot compare behavior with v.12 as runboat has issues

**Comment by StefanRijnhart** [comment]
That would be the standard document layout popout that you need to go through on every new Odoo database. What happens if you go through that wizard and then try again?

**Comment by francesco-ooops** [comment]
> That would be the standard document layout popout that you need to go through on every new Odoo database. 

Which option should be selected for the module to work?

> What happens if you go through that wizard and then try again?

"The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application."

**Comment by StefanRijnhart** [comment]
Any option will do. It's just to determine the basic layout for other kinds of reports.

I tried printing the labels on a fresh install, and I do not encounter an error: 
image

**Review by francesco-ooops — APPROVED** [review]
good for me then, maybe it's due to the runboat not building completely in the first place

who can we ask for merge?

**Comment by StefanRijnhart** [comment]
Ah, you installed the module manually on runboat. In that case I can fetch the error from the logs there, and it's

```
Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/http.py", line 806, in dispatch
    r = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 347, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/mnt/data/odoo-addons-dir/report_xml/controllers/main.py", line 95, in report_download
    return super().report_download(data, token, context)
  File "/opt/odoo/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo-venv/lib/python3.6/site-packages/werkzeug/serving.py", line 306, in run_wsgi
    execute(self.server.app)
  File "/opt/odoo-venv/lib/python3.6/site-packages/werkzeug/serving.py", line 294, in execute
    application_iter = app(environ, start_response)
  File "/opt/odoo/odoo/service/server.py", line 441, in app
    return self.app(e, s)
  File "/opt/odoo/odoo/service/wsgi_server.py", line 113, in application
    return application_unproxied(environ, start_response)
  File "/opt/odoo/odoo/service/wsgi_server.py", line 88, in application_unproxied
    result = odoo.http.root(environ, start_response)
  File "/opt/odoo/odoo/http.py", line 1328, in __call__
    return self.dispatch(environ, start_response)
  File "/opt/odoo/odoo/http.py", line 1294, in __call__
    return self.app(environ, start_wrapped)
  File "/opt/odoo-venv/lib/python3.6/site-packages/werkzeug/middleware/shared_data.py", line 220, in __call__
    return self.app(environ, start_response)
  File "/opt/odoo/odoo/http.py", line 1518, in dispatch
    result = ir_http._dispatch()
  File "/opt/odoo/addons/auth_signup/models/ir_http.py", line 19, in _dispatch
    return super(Http, cls)._dispatch()
  File "/opt/odoo/addons/web_editor/models/ir_http.py", line 21, in _dispatch
    return super(IrHttp, cls)._dispatch()
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 241, in _dispatch
    return cls._handle_exception(e)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 209, in _handle_exception
    return request._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 744, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: report_download() takes 3 positional arguments but 4 were given - - -
```

Let me try to fix it here.

**Comment by StefanRijnhart** [comment]
@francesco-ooops I Just checked on runboat, after manual installation of the module and selection of the company reporting style, you can now print the demo labels from the module in this PR.

**Comment by francesco-ooops** [comment]
> @francesco-ooops I Just checked on runboat, after manual installation of the module and selection of the company reporting style, you can now print the demo labels from the module in this PR.

confirmed, LGTM now!

**Comment by francesco-ooops** [comment]
@OCA/reporting-engine-maintainers can we proceed with merge?

**Comment by toan-syncoria** [comment]
Do we have ver 15?

**Pull request merged**
